sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/StringUtilsSuite.scala

```diff
@@ -98,4 +98,12 @@ class StringUtilsSuite extends SparkFunSuite {
     assert(checkLimit("1234567"))
     assert(checkLimit("1234567890"))
   }
+
+  test("StringConcat doesn't overflow on many inputs") {
```

Comment:
nit: could you please add the prefix: "SPARK-31916: " ?

Reply:
@kiszk Thanks.. will do.